Experimental/telemetry

ChangJoseph opened the pull request:
# Main changes
- Made LoggedData.java that handles each codex and does the codex to CSV logging
- Made CodexNetworkTablesParser.java to translate NT to the local Codex, construct the NT (codex) name, and create the CSV header/logs and CSV file path
- Changed CodexNetworkTables.java to be able to send multiple codexes with the same enumeration to NT
- Changed Data.java to register codexes and send them to their corresponding NTs
# Minor changes
- Changed CodexNetworkTables.java NT hash method from EnumUtils.class to the String's hashCode

farissyed committed:
Commented out old dependencies

DanielTolbert committed:
Merge branch 'config/newHFR' into experimental/telemetry

DanielTolbert committed:
Fixed build.gradlew issues and updated Codex imports

ChangJoseph committed:
Data NT Testing

Made debug variable for gyro entry on NT

ChangJoseph committed:
Data NT Debugging

Made debug variable for gyro entry on NT

ChangJoseph committed:
NT & Codex Test Methods

codex gyro yaw getter/setter; need NT yaw getter/setter *note - method names are WIP*

ChangJoseph committed:
Merge remote-tracking branch 'origin/experimental/telemetry' into experimental/telemetry

ChangJoseph committed:
Cleaned code

Added comments; changed method name; removed extraneous code

ChangJoseph committed:
Merge remote-tracking branch 'origin/master' into experimental/telemetry

ChangJoseph committed:
Fixed master merge conflicts

Added backup CSVLogger

ChangJoseph committed:
Finalized and Tested NT to Codex

CSVLogger is now deprecated, LoggedData will hold all used codexes, deleted useless Data methods, Added CodexNetworkTablesParser to translate NetworkTable to Codex

ChangJoseph committed:
Made test methods for CSV

ChangJoseph committed:
Finished base logging methods

ChangJoseph committed:
Merge remote-tracking branch 'origin/master' into experimental/telemetry

ChangJoseph committed:
Testing method for CSV logging created

build.gradle change -> GradleRIO version 2019.2.1

ChangJoseph committed:
Fixed test methods

Moved from LoggedData to Robot in order to get robot time sent data

ChangJoseph committed:
Started multiple enum instances support

Format for multi-enum codices: 'NAME-ENUM'
Removed unneeded dependencies
Moved CSV logging from robot to LoggedData's main

ChangJoseph committed:
Merge remote-tracking branch 'origin/master' into experimental/telemetry

ChangJoseph committed:
Finished multi-enum logging, cleaned, add doc

Removed need for Class<E> parameter for CodexNetworkTablesParser by using codex.meta().getEnum()
Cleaned the code:
- changed method ordering
- clarified documentation

ChangJoseph committed:
Changed structure of logging and finished main CSV logging

Need to test when receiving actual data, need to merge, need to test enumeration multi-instance

ChangJoseph committed:
Merge remote-tracking branch 'origin/master' into experimental/telemetry

ChangJoseph committed:
Merge remote-tracking branch 'origin/master' into experimental/telemetry

ChangJoseph committed:
Fixed codex multi-enum instance

ChangJoseph committed:
Deleted CSVLogger.java

ChangJoseph committed:
version 2019.2.1 -> 2019.1.1

StephenWelch commented on common/src/main/java/us/ilite/common/LoggedData.java:
```diff
@@ -0,0 +1,148 @@
+    }
+
+    //For testing purposes
+    public static void main(String[] args) {
```
This thread along with any code for writing Codexes should be moved to a separate `DataLogger` class that extends `Thread`. `LoggedData` itself should be replaced by the `Data` class from the `robot` project.

ChangJoseph replied:
What if I keep the *Data* class and just rename it to *RobotData* (and keep *LoggedData*)? This is because *Data* is needed in *robot\* to send the robot's codexes to NTs and has some fields that are referenced in other robot classes.

anseljohn commented on common/src/main/java/us/ilite/common/io/CodexNetworkTables.java:
```diff
@@ -64,6 +82,34 @@
+    }
+  }
+
   private CodexNetworkTables() {
```
You can get rid of this constructor

anseljohn replied:
I see what it is for now...

anseljohn commented on common/src/main/java/us/ilite/common/LoggedData.java:
```diff
@@ -0,0 +1,148 @@
+    private void handleCreation(File pFile) throws IOException {
+        //Makes every folder before the file if the CSV's parent folder doesn't exist
+        if(!pFile.getParentFile().exists()) pFile.getParentFile().mkdirs();
+
```
It'd be a little better just for readability to put if statements on separate lines as the calls inside the statements

ChangJoseph replied:
I got the comments explaining the *if statement* for this reason. Do you think I should still separate it into multiple lines?

anseljohn replied:
yes

ChangJoseph committed:
Renamed Data, made DataLogger, cleaned handleCreation

Data -> RobotData
DataLogger runs the csv logging
handleCreation now has separated if statements

ChangJoseph committed:
Merge remote-tracking branch 'origin/master' into experimental/telemetry

ChangJoseph committed:
Removing RobotData part1

Commit:
Fix imports

ChangJoseph committed:
Merge remote-tracking branch 'origin/master' into experimental/telemetry
